src/modules/hip/kernel/color_temperature.hpp

```diff
@@ -0,0 +1,233 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+__device__ void color_temperature_hip_compute(uchar *srcPtr, d_float24 *pix_f24, float4 *adjustmentValue_f4)
+{
```

Comment:
why is the srcPtr passed to this function?

Reply:
We had a discussion long ago on this, I think we should discuss these on a separate short call.

Reply:
Sure Rajy we will check performance with Templated version as discussed.

Reply:
Performance numbers are similar to using templated function for each datatype with minor differences due to noise. Modified them accordingly.